Loop/Managers/DeviceAlert/UserNotificationDeviceAlertHandler.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  UserNotificationAlertHandler.swift
+//  LoopKit
+//
+//  Created by Rick Pasetto on 4/9/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import LoopKit
+import UserNotifications
+
+class UserNotificationDeviceAlertHandler: DeviceAlertHandler {
+    
+    let alertInBackgroundOnly: Bool
+    let isAppInBackgroundFunc: () -> Bool
+    let userNotificationCenter: UNUserNotificationCenter = UNUserNotificationCenter.current()
+    
+    init(alertInBackgroundOnly: Bool = true, isAppInBackgroundFunc: @escaping () -> Bool) {
+        self.alertInBackgroundOnly = alertInBackgroundOnly
+        self.isAppInBackgroundFunc = isAppInBackgroundFunc
+    }
+        
+    func issueAlert(_ alert: DeviceAlert) {
+        DispatchQueue.main.async {
+            if self.alertInBackgroundOnly && self.isAppInBackgroundFunc() || !self.alertInBackgroundOnly {
+                if let request = alert.asUserNotificationRequest() {
+                    self.userNotificationCenter.add(request)
+                    // For now, UserNotifications do not not acknowledge...not yet at least
+                }
+            }
+        }
+    }
+    
+    func removePendingAlert(identifier: DeviceAlert.Identifier) {
+        userNotificationCenter.removePendingNotificationRequests(withIdentifiers: [identifier.value])
+    }
+    
+    func removeDeliveredAlert(identifier: DeviceAlert.Identifier) {
+        userNotificationCenter.removeDeliveredNotifications(withIdentifiers: [identifier.value])
+    }
+}
+
+public extension DeviceAlert {
+    
+    fileprivate func asUserNotificationRequest() -> UNNotificationRequest? {
+        guard let uncontent = getUserNotificationContent() else {
+            return nil
+        }
+        return UNNotificationRequest(identifier: identifier.value,
+                                     content: uncontent,
+                                     trigger: trigger.asUserNotificationTrigger())
+    }
+    
+    private func getUserNotificationContent() -> UNNotificationContent? {
+        guard let content = backgroundContent else {
+            return nil
+        }
+        let userNotificationContent = UNMutableNotificationContent()
+        userNotificationContent.title = content.title
+        userNotificationContent.body = content.body
+        userNotificationContent.sound = content.isCritical ? .defaultCritical : .default
```

Comment:
We do need to deal with custom sounds per the Dexcom integration. We don't have the actual sound files yet, afaik, but this facility will need to be revamped to support custom sounds. That might be out of the scope of this ticket. If so, do we have a ticket covering this? We know we need it.

Reply:
https://tidepool.atlassian.net/browse/LOOP-1179